source/extensions/filters/network/thrift_proxy/conn_manager.cc

```diff
@@ -404,6 +412,14 @@ void ConnectionManager::ActiveRpc::finalizeRequest() {
 
   parent_.stats_.request_.inc();
 
+  parent_.accumulated_requests_++;
+  if (parent_.config_.maxRequestsPerConnection() > 0 &&
+      parent_.accumulated_requests_ >= parent_.config_.maxRequestsPerConnection()) {
```

Comment:
this should actually be:

```
parent_.accumulated_requests_ > parent_.config_.maxRequestsPerConnection()
```

if I set this to 50 requests... I want 50 requests to go through.